frontend/src/types/Mesh.ts

```diff
@@ -35,22 +37,81 @@ export enum MeshInfraType {
 }
 
 export enum MeshNodeType {
-  BOX = 'box',
-  INFRA = 'infra'
+  Box = 'box',
+  Infra = 'infra'
 }
 
 export type MeshNodeHealthData = string;
 
-// Node data expected from server
-export interface MeshNodeData {
+export interface IstiodNodeData extends BaseNodeData {
+  infraData: ControlPlane;
+  infraType: MeshInfraType.ISTIOD;
+}
+
+export interface NamespaceNodeData extends BaseNodeData {
+  infraType: MeshInfraType.NAMESPACE;
+}
+
+export interface ClusterNodeData extends BaseNodeData {
+  infraData: MeshCluster;
+  infraType: MeshInfraType.CLUSTER;
+}
+
+export interface DataPlaneNodeData extends BaseNodeData {
+  infraData: NamespaceInfo[];
+  infraType: MeshInfraType.DATAPLANE;
+}
+
+export interface GrafanaNodeData extends BaseNodeData {
+  // Grafana node data is the raw grafana config. We don't actually care about what
+  // each field is since we just display the whole config in the side panel as is.
+  infraData: any;
```

Comment:
If possible, it is better to use `unknown`, which is the type-safe counterpart of any.

```suggestion
  infraData: unknown;
```